packages/neon-ledger/src/main.ts

```diff
@@ -90,26 +137,32 @@ export async function getPublicKey(
 /**
  * Requests the device to sign a message using the NEO application.
  * @param ledger - Ledger instance
- * @param hex - message to sign as a hexstring.
+ * @param payload - message to sign as a hexstring.
  * @param bip44String - BIP44 string (40 bytes)
+ * @param networkMagic - MainNet, TestNet or custom network magic number
  * @returns Signature as a hexstring (64 bytes)
  */
 export async function getSignature(
   ledger: Transport,
-  hex: string,
-  bip44String: string
+  payload: string,
+  bip44String: string,
+  networkMagic: number
 ): Promise<string> {
-  const payload = hex + bip44String;
+  await sendDataToSign(ledger, bip44String, 0);
+  await sendDataToSign(ledger, u.num2hexstring(networkMagic, 4, true), 1);
+
   const chunks = payload.match(/.{1,510}/g) || [];
   try {
     for (let i = 0; i < chunks.length - 1; i++) {
-      await appendDataForSignature(ledger, chunks[i]);
+      await sendDataToSign(ledger, chunks[i], 2 + i);
     }
-    const response = await finalizeDataForSignature(
+    const response = await sendDataToSign(
       ledger,
-      chunks[chunks.length - 1]
+      chunks[chunks.length - 1],
+      2 + chunks.length,
+      true
     );
-    if (response.readUIntBE(0, 2) === ErrorCode.VALID_STATUS) {
+    if (response.readUIntBE(0, 2) === StatusWord.OK) {
```

Comment:
On a side note, what is actually the correct status that should be returned here? My understanding was that the ledger will keep returning OK as it accepts new data for signature and if we still see OK at this point, it means something weird happened. If we know the correct return value, we can check for that instead.

Reply:
Every `apdu` send to the device will get at least 2 bytes of response in the form of a status word. `0x9000` just means the apdu was processed as expected, otherwise we get some other status word like `0x6e01` for `APP_CLOSED`. It can also reply with "data + status word". It is up to us to decide how the protocol works. What you describe could very well have been the behaviour of the old app. 

For the new app there is no such thing as "OK" means we can receive more data. The new "protocol" for signing an app is
-> send bip44 path
<- `0x9000`
-> send network magic
<- `0x9000`
-> send unsigned data part of the tx
<- receive signature + `0x9000`

Having said that I realize that piece of code is no longer applicable. I removed it

Reply:
upon failure, I suppose the ledger bridge should throw an error which would invoke the try-catch. Can you add a comment explaining the latest edit? Probably something along the lines of `Expected signature + 2 status bytes to be returned here upon completion`